### ISSUE TITLE
rename service_number field everywhere

### ISSUE THREAD

**Issue by tballmsft** [microsoft/jacdac-ts#86]
I suggest we rename service_number to service_index, which more accurately reflects that the value of the field is an index into a device's advertised list of services.

**Comment by jamesadevine** [comment]
I think on one of our calls we opted for `service_instance` rather than `service_index`

**Comment by pelikhan** [comment]
Index is more accurate IMHO

________________________________

**Comment by jamesadevine** [comment]
Yeah, I have no issue with either, just reminding us what we decided on a call.

**Comment by tballmsft** [comment]
Sorry, I don't remember any of that. I prefer "index" over "instance".

**Comment by jamesadevine** [comment]
Yeah, me too! I think it was called that at some point...

**Comment by tballmsft** [comment]
one strange thing is  the use of `JD_SERVICE_NUMBER_CRC_ACK` as a service number/index.   Any reason this couldn't be a command in the control service?

**Comment by jamesadevine** [comment]
@mmoskal

**Comment by tballmsft** [comment]
IMO, we should adhere to the service abstraction interface as much as possible,

**Comment by pelikhan** [comment]
So "service_index"?

**Comment by tballmsft** [comment]
yes, "service_number" -> "service_index" everywhere.
